src/lib.rs

```diff
@@ -410,6 +416,82 @@ impl Contract {
+        self.available_addons.get(&id)
+    }
+
+    pub fn get_all_addons(&self) -> Vec<AddOn> {
+        self.available_addons.iter().map(|(_id, add_on)| add_on).collect()
+    }
+
+    // Only the contract admin and DevHub moderators
+    pub fn create_addon(&mut self, addon: AddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can create new add-ons");
+        }
+        if self.get_addon(addon.id.to_owned()).is_some() {
+            panic!("Add-on with this id already exists");
+        }
+        addon.validate();
+        self.available_addons.insert(&addon.id.clone(), &addon);
+    }
+
+    // ONLY FOR TESTING
+    pub fn delete_addon(&mut self, id: AddOnId) {
+        // Also delete from communities
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can delete add-ons");
+        }
+        let addon = self
+            .get_addon(id.clone())
+            .expect(&format!("Add-on with id `{}` does not exist", id))
+            .clone();
+
+        self.available_addons.remove(&addon.id);
+    }
+
+    pub fn update_addon(&mut self, addon: AddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can edit add-ons");
+        }
+        self.available_addons.insert(&addon.id.clone(), &addon);
+    }
+
+    pub fn get_community_addons(&self, handle: CommunityHandle) -> Vec<CommunityAddOn> {
+        let community = self
+            .get_community(handle.clone())
+            .expect(format!("Community not found with handle `{}`", handle).as_str());
+        community.addons
+    }
+
+    pub fn set_community_addons(&mut self, handle: CommunityHandle, addons: Vec<CommunityAddOn>) {
+        let mut community = self
+            .get_community(handle.clone())
+            .expect(format!("Community not found with handle `{}`", handle).as_str());
+        community.addons = addons;
+        self.update_community(handle, community);
+    }
+
+    // To add or update parameters set by the configurator widget
+    pub fn set_community_addon(self, handle: CommunityHandle, community_addon: CommunityAddOn) {
+        let mut community = self
+            .get_community(handle.clone())
+            .expect(format!("Community not found with handle `{}`", handle).as_str());
+        if let Some(index) =
+            community.addons.iter().position(|current| current.id == community_addon.id)
+        {
+            community.addons[index] = community_addon;
+        } else {
+            community.addons.push(community_addon);
+        }
+    }
```

Comment:
I believe you need to call `update_community` to persist the changes:

```suggestion
        self.update_community(handle, community);
    }
```